libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/SfapApiChatGenerationsRequestBody.kt

```diff
@@ -0,0 +1,115 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.rest
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+
+/**
+ * Models a `sfap_api` `chat-generations` endpoint request.
+ * See https://developer.salesforce.com/docs/einstein/genai/references/models-api?meta=generateChat
+ *
+ * @param messages The request messages parameter value
+ * @param localization The request localization parameter value
+ * @param tags The request tags parameter value
+ */
+@Serializable
+data class SfapApiChatGenerationsRequestBody(
+    val messages: Array<Message>,
+    val localization: Localization,
+    val tags: Tags
+) {
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as SfapApiChatGenerationsRequestBody
+
+        if (!messages.contentEquals(other.messages)) return false
+        if (localization != other.localization) return false
+        if (tags != other.tags) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = messages.contentHashCode()
+        result = 31 * result + localization.hashCode()
+        result = 31 * result + tags.hashCode()
+        return result
+    }
+
+
+    /**
+     * Encodes this request body to JSON text.
+     * @return This request body as JSON text
+     */
+    fun toJson() = Json.encodeToString(serializer(), this)
+
+    @Serializable
+    data class Message(
+        val role: String,
+        val content: String
+    )
+
+    @Serializable
+    data class Localization(
+        val defaultLocale: String,
+        val inputLocales: Array<Locale>,
+        val expectedLocales: Array<String>
+    ) {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as Localization
+
+            if (defaultLocale != other.defaultLocale) return false
+            if (!inputLocales.contentEquals(other.inputLocales)) return false
+            if (!expectedLocales.contentEquals(other.expectedLocales)) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = defaultLocale.hashCode()
+            result = 31 * result + inputLocales.contentHashCode()
+            result = 31 * result + expectedLocales.contentHashCode()
+            return result
+        }
+    }
+
+    @Serializable
+    data class Locale(
+        val locale: String,
+        val probability: Double
+    )
+
+    @Serializable
+    class Tags
```

Comment:
@wmathurin - The documentation doesn't have any content for this `Tags` object.  Is that a concern? Or, since I left unknown keys on for requests is it up to the app to subclass this type and add some arbitrary content to it?

Reply:
The doc is indeed very vague

> Entries used by the Models API for non-generative purposes and entries used by the client for free-form data." 

See https://developer.salesforce.com/docs/einstein/genai/references/models-api?meta=type%3ATags

Assuming it's a map of String to String, what would the code look like if you try to provide them?

Reply:
I tried a couple options this morning, but I couldn't get Kotlin Serialization to give me the developer experience I was thinking of.  🤔. I'll try a couple other options after the holiday and see what I can whip up.